src/itest/java/uk/gov/companieshouse/company/metrics/steps/AppointmentSteps.java

```diff
@@ -0,0 +1,195 @@
+    public void resourceChangedDataMessageExistsOnMainTopic(String companyNumber, String topic) {
+        ResourceChangedData messageData = testSupport.createResourceChangedMessageAppointments(
+                COMPANY_METRICS_RECALCULATE_URI, companyNumber, CONTEXT.get(EVENT_TYPE).toString());
+        currentCompanyNumber = companyNumber;
+
+        sendKafkaMessage(topic, messageData);
+    }
+
+    @Given("A message for {string} with invalid appointment entity exists on the {string} kafka topic")
+    public void resourceChangedDataMessageWithInvalidAppointmentEntityExistsOnTopic(String companyNumber, String topic) {
+
+        ResourceChangedData messageData = testSupport.createResourceChangedMessageInvalidAppointments(
+                COMPANY_METRICS_RECALCULATE_URI, companyNumber);
+        currentCompanyNumber = companyNumber;
+
+        sendKafkaMessage(topic, messageData);
+    }
+
+    @Given("An invalid message exists on the {string} kafka topic")
+    public void invalidMessageExistsOnMainKafkaTopic(String topic) {
+        kafkaTemplate.send(topic, "invalidMessage");
+        kafkaTemplate.flush();
+    }
+
+    @Then("The message should be placed on to {string} kafka topic")
+    public void messageShouldBePlacedOntoTopic(String topic) {
+        ConsumerRecord<String, Object> singleRecord =
+                KafkaTestUtils.getSingleRecord(kafkaConsumer,topic, 5000L);
+
+        String recordTopic = singleRecord.topic();
+        assertThat(recordTopic).isEqualTo(topic);
+        assertThat(singleRecord.value()).isNotNull();
+    }
+
+    @Then("A non-retryable exception should be thrown when consuming from {string}")
+    public void aNonRetryableExceptionIsThrown(String topic) {
+        byte[] invalidMessage = "invalidMessage".getBytes();
+        Exception exception = assertThrows(NonRetryableErrorException.class, () -> {
+            resourceChangedDataDeserializer.deserialize(topic, invalidMessage);
+        });
+
+        String expectedMessage = "Malformed data";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @And("The event type is {string}")
+    public void eventTypeIsSetToChanged(String eventType){
+        CONTEXT.set(EVENT_TYPE, eventType);
+    }
+
+    @When("The message is consumed")
+    public void messageSuccessfullyConsumed() throws InterruptedException {
+        assertThat(resettableCountDownLatch.getCountDownLatch().await(5, TimeUnit.SECONDS)).isTrue();
+    }
+
+    @And("A request is sent to the Company Metrics Recalculate endpoint")
+    public void requestSentToCompanyMetricsRecalculateEndpoint() {
+        List<ServeEvent> serverEvents = checkServeEvents();
+        assertMetricsApiSuccessfullyInvoked(serverEvents);
+    }
+
+    @Given("The consumer has been configured with api key without internal app privileges for {string}")
+    public void consumerBeenConfiguredAsUnauthorised(String companyNumber) {
```

Comment:
Is `companyNumber` required here? Should it be removed from the parameters/feature file if not needed?